channels/pkg/cmd/apply_channel.go

```diff
@@ -273,3 +364,20 @@ func buildMenu(kubernetesVersion semver.Version, args []string, localFiles bool)
 	}
 	return menu, nil
 }
+
+func deleteAddon(ctx context.Context, k8sClient kubernetes.Interface, pruner *channels.Pruner, key string) error {
+	parts := strings.Split(key, ":")
```

Comment:
TBH this scares me a little.  We should have a parsing function, and that function should be strict.  Not a problem for this PR, but I might add such a function.